pac4j-core/src/main/java/org/pac4j/core/context/session/J2ESessionStore.java

```diff
@@ -50,17 +51,13 @@ public boolean destroySession(final J2EContext context) {
     }
 
     @Override
-    public Object getTrackableSession(final J2EContext context) {
-        return getHttpSession(context);
+    public Optional getTrackableSession(final J2EContext context) {
+        return Optional.of(getHttpSession(context));
     }
 
     @Override
-    public SessionStore<J2EContext> buildFromTrackableSession(final J2EContext context, final Object trackableSession) {
-        if (trackableSession != null) {
-            return new J2EProvidedSessionStore((HttpSession) trackableSession);
-        } else {
-            return null;
-        }
+    public Optional<SessionStore<J2EContext>> buildFromTrackableSession(final J2EContext context, final Object trackableSession) {
```

Comment:
shouldn't trackableSession be an Optional?

Either they are both or none are for me… actually, I don't think we really need to introduce Optional here, it is the caller that should be careful of not calling this with a null value!